src/utils/hooks.ts

```diff
@@ -1,16 +1,95 @@
-import { graphql, useStaticQuery } from 'gatsby'
-
-export const useSiteMetadata = () => {
-  const { site } = useStaticQuery(
-    graphql`
-      query {
-        site {
-          siteMetadata {
-            title
-          }
-        }
-      }
-    `,
-  )
-  return site.siteMetadata
+import { useCallback, useEffect, useLayoutEffect, useState } from 'react'
+
+export const useDimensions = () => {
+  const [dimensions, setDimensions] = useState({ width: 0 })
+  const [node, setNode] = useState(null)
+
+  const ref = useCallback(node => {
+    setNode(node)
+  }, [])
+
+  useLayoutEffect(() => {
+    if (node == null) {
+      return
+    }
+    const measure = () =>
+      window.requestAnimationFrame(() => {
+        // @ts-ignore
+        const rect = node.getBoundingClientRect()
+        setDimensions({ width: rect.width })
+      })
+    measure()
+
+    window.addEventListener('resize', measure)
+    window.addEventListener('scroll', measure)
+
+    return () => {
+      window.removeEventListener('resize', measure)
+      window.removeEventListener('scroll', measure)
+    }
+  }, [node])
+  return [ref, dimensions, node]
+}
+
+export const useMedia = (queries, values, defaultValue) => {
+  // Array containing a media query list for each query
+  const mediaQueryLists =
+    typeof window !== 'undefined' ? queries.map(q => window.matchMedia(q)) : []
+
+  // Function that gets value based on matching media query
+  // eslint-disable-next-line
+  const getValue = () => {
+    // Get index of first media query that matches
+    const index = mediaQueryLists.findIndex(mql => mql.matches)
+    // Return related value or defaultValue if none
+    return typeof values[index] !== 'undefined' ? values[index] : defaultValue
+  }
+
+  // State and setter for matched value
+  const [value, setValue] = useState(getValue)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Expected indentation of 4 spaces but found 2. (@typescript-eslint/indent)](https://app.codacy.com/app/jdstuart/jesses.io/pullRequest?prid=4576429)